src/main/java/Duke.java

```diff
@@ -1,10 +1,200 @@
+     * it is assumed that the input has correct structure)
+     */
+    private static void handleCommand(String command) throws DukeException {
+        if (command.equals("bye")) {
+            exit();
+        } else if (command.equals("list")) {
+            printList();
+        } else if (command.split(" ")[0].equals("done")) {
+            int index = Integer.parseInt(command.split(" ")[1]);
+            done(tasks.get(index - 1));
+            try {
+                writeToFile();
+            } catch (IOException e) {
+                System.out.println("Oops! " + e.getMessage());
+            }
+        } else if (command.split(" ")[0].equals("delete")) {
+            delete(Integer.parseInt(command.split(" ")[1]));
+            try {
+                writeToFile();
+            } catch (IOException e) {
+                System.out.println("Oops! " + e.getMessage());
+            }
+        } else if (command.split(" ")[0].equals("todo")) {
+            String taskcommand = command.replace("todo", "");
+            if (!taskcommand.equals("")) {
+                add(new ToDo(taskcommand));
+            } else {
+                throw new DukeException("EmptyToDo");
+            }
+            try {
+                writeToFile();
+            } catch (IOException e) {
+                System.out.println("Oops! " + e.getMessage());
+            }
+        } else if (command.split(" ")[0].equals("deadline")) {
+            String taskcommand = command.split("/")[0].replace("deadline", "");
+            if (!taskcommand.equals("")) {
+                String time = command.split("/")[1].replace("by ", "");
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+                add(new Deadline(taskcommand, LocalDateTime.parse(time, formatter)));
+            } else {
+                throw new DukeException("EmptyDeadline");
+            }
+            try {
+                writeToFile();
+            } catch (IOException e) {
+                System.out.println("Oops! " + e.getMessage());
+            }
+        } else if (command.split(" ")[0].equals("event")) {
+            String taskcommand = command.split("/")[0].replace("event", "");
+            if (!taskcommand.equals("")) {
+                String time = command.split("/")[1].replace("at ", "");
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+                add(new Deadline(taskcommand, LocalDateTime.parse(time, formatter)));
+            } else {
+                throw new DukeException("EmptyEvent");
+            }
+            try {
+                writeToFile();
+            } catch (IOException e) {
+                System.out.println("Oops! " + e.getMessage());
+            }
+        } else {
+            throw new DukeException("invalid");
+        }
+    }
+
+    private static void readFile() throws FileNotFoundException {
```

Comment:
As per iP task on code quality, should this function be accompanied with a JavaDoc?